src/lib/admin-auth.ts

```
@@ -1,149 +1,165 @@
-    
-    const passwordHash = process.env.ADMIN_PASSWORD_HASH;
-    if (!passwordHash) {
-      console.error('ADMIN_PASSWORD_HASH not configured');
+  static async isAuthenticatedFromRequest(
+    request: NextRequest
+  ): Promise<boolean> {
+    try {
+      const token = request.cookies.get("admin-token")?.value;
+      if (!token) return false;
+
+      const user = await this.verifyToken(token);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)